src/physics/utils/phys_comp.F90

```diff
@@ -245,60 +202,76 @@ subroutine phys_run2(dtime_phys, cam_runtime_opts, phys_state, phys_tend,  &
            read_initialized_variables=use_init_variables)
 
       ! Initialize the physics time step
-      call cam_ccpp_physics_timestep_initial(phys_suite_name, dtime_phys,     &
-           errmsg, errcode)
+      call cam_ccpp_physics_timestep_initial(phys_suite_name)
       if (errcode /= 0) then
          call endrun('cam_ccpp_physics_timestep_initial: '//trim(errmsg))
       end if
 
-      ! Threading vars
-      col_start = 1
-      col_end = columns_on_task
+   end subroutine phys_timestep_init
+
+   subroutine phys_run1()
+      use cam_ccpp_cap,   only: cam_ccpp_physics_run
+      use cam_abortutils, only: endrun
 
-      ! Run CCPP suite
-      do part_ind = 1, size(suite_parts, 1)
-         call cam_ccpp_physics_run(phys_suite_name, suite_parts(part_ind),    &
-              col_start, col_end, dtime_phys, errmsg, errcode)
+      ! Run before coupler group if it exists
+      if (any('physics_before_coupler' == suite_parts)) then
+         call cam_ccpp_physics_run(phys_suite_name, 'physics_before_coupler')
          if (errcode /= 0) then
             call endrun('cam_ccpp_physics_run: '//trim(errmsg))
          end if
-      end do
```

Comment:
Just pointing out that this old logic had the ability to run numerous groups in the order they were presented in the SDF.  I realize that it didn't have the ability to indicate where the dycore needed to be called, which is what prompted this change.  It would be nice to not have our host model depend on having specific group names, but that is what we need at this point in the development process.  It would be nice to strive for a way to have "call dycore now" capabilities in any SDF.  At that point, we could just call the groups in the order they appear in the SDF and we could resurrect this loop.  Perhaps something for "future us".

Reply:
yes, agreed that it would be nice to have a way to specify coupling/dycore in the SDF

Reply:
I've added this idea to issue #73 and given it a "wishlist" tag, so we can remember to try and tackle it in the future (also feel free to add to that issue if I missed anything or didn't describe the idea correctly).  Thanks!